src/core/qgsstoredexpressionmanager.h

```diff
@@ -0,0 +1,158 @@
+/***************************************************************************
+                          qgsstoredexpressionmanager.h
+
+ These classes store and control the management and execution of actions
+ associated with a particular Qgis layer. Actions are defined to be
+ external programs that are run with user-specified inputs that can
+ depend on the contents of layer attributes.
+
+                             -------------------
+    begin                : August 2019
+    copyright            : (C) 2019 David Signer
+    email                : david at opengis dot ch
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSSTOREDEXPRESSIONMANAGER_H
+#define QGSSTOREDEXPRESSIONMANAGER_H
+
+#include "qgis_core.h"
+#include <QString>
+#include <QObject>
+#include <QUuid>
+
+
+#ifdef SIP_RUN
+% ModuleHeaderCode
+#include <qgsstoredexpressionmanager.h>
+% End
+#endif
+
+class QDomNode;
+class QDomDocument;
+
+struct CORE_EXPORT QgsStoredExpression
+{
+
+  /* leads to trouble with sip...
+  QgsStoredExpression() = default;
+
+  QgsStoredExpression( QString name, QString expression )
+    : id( QUuid::createUuid() ),
+      name( name ),
+      expression( expression )
+  {}
+  */
+
+  QUuid id;
```

Comment:
I think instead of using raw QUUid's here, it would be nicer to use a QString in the public API. You can use QUuid in the background to generate initial values for the strings, but only expose them publicly as strings.